Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/StreamingImageAsset.cpp

```diff
@@ -13,6 +13,119 @@
 
 namespace AZ
 {
+    namespace Internal
+    {
+        template <AZ::RHI::Format>
+        float RetrieveFloatValue(const AZ::u8* mem, size_t index)
+        {
+            static_assert(false, "Unsupported pixel format");
+        }
+
+        template <AZ::RHI::Format>
+        AZ::u32 RetrieveUintValue(const AZ::u8* mem, size_t index)
+        {
+            static_assert(false, "Unsupported pixel format");
+        }
+
+        template <AZ::RHI::Format>
+        AZ::s32 RetrieveIntValue(const AZ::u8* mem, size_t index)
+        {
+            static_assert(false, "Unsupported pixel format");
+        }
+
+        template <>
+        float RetrieveFloatValue<AZ::RHI::Format::Unknown>([[maybe_unused]] const AZ::u8* mem, [[maybe_unused]] size_t index)
+        {
+            return 0.0f;
+        }
+
+        template <>
+        AZ::u32 RetrieveUintValue<AZ::RHI::Format::Unknown>([[maybe_unused]] const AZ::u8* mem, [[maybe_unused]] size_t index)
+        {
+            return 0;
+        }
+
+        template <>
+        AZ::s32 RetrieveIntValue<AZ::RHI::Format::Unknown>([[maybe_unused]] const AZ::u8* mem, [[maybe_unused]] size_t index)
+        {
+            return 0;
+        }
+
+        float RetrieveFloatValue(const AZ::u8* mem, size_t index, AZ::RHI::Format format)
+        {
+            switch (format)
+            {
+            case AZ::RHI::Format::R8_UNORM:
+            case AZ::RHI::Format::R8_SNORM:
+            case AZ::RHI::Format::A8_UNORM:
+            {
+                return mem[index] / static_cast<float>(std::numeric_limits<AZ::u8>::max());
+            }
+            case AZ::RHI::Format::R16_FLOAT:
+            case AZ::RHI::Format::D16_UNORM:
+            case AZ::RHI::Format::R16_UNORM:
+            case AZ::RHI::Format::R16_SNORM:
```

Comment:
These are all subtly different. What you have is correct for `D16_UNORM` and `R16_UNORM`. 

For `R16_FLOAT`, the bits need to be interpreted as a half-float, then unpacked into a full float. There's some example code in Gems\Atom\Asset\ImageProcessingAtom\Code\Source\Processing\PixelFormatInfo.h. I put in a request several months ago to get first class half-float conversion support [here ](https://github.com/o3de/o3de/issues/3857)

For `R16_SNORM`, mem should be treated as a s16 then scaled from it's min/max to -1 to 1.

Reply:
Thanks! I will update them. For `R16_FLOAT`, the `PixelFormatInfo.h` is only available for Editor, so I'll just copy that code for now and tag on that issue so that it gets replaced when we provide first class support for half-float conversion.